addons/stock/models/stock_move_line.py

```diff
@@ -144,11 +144,32 @@ def _onchange_serial_number(self):
                     counter = Counter([line.lot_name for line in move_lines_to_check])
                     if counter.get(self.lot_name) and counter[self.lot_name] > 1:
                         message = _('You cannot use the same serial number twice. Please correct the serial numbers encoded.')
+                    elif not self.lot_id:
+                        counter = self.env['stock.production.lot'].search_count([
+                            ('company_id', '=', self.company_id.id),
+                            ('product_id', '=', self.product_id.id),
+                            ('name', '=', self.lot_name),
+                        ])
+                        if counter > 0:
+                            message = _('Existing Serial number (%s). Please correct the serial number encoded.') % self.lot_name
                 elif self.lot_id:
                     counter = Counter([line.lot_id.id for line in move_lines_to_check])
                     if counter.get(self.lot_id.id) and counter[self.lot_id.id] > 1:
                         message = _('You cannot use the same serial number twice. Please correct the serial numbers encoded.')
-
+                    else:
+                        counter = self.env['stock.move.line'].search_count([
```

Comment:
I'm not sure about this search (cc @amoyaux). We have a lot_id but no lot_name. which means we will not create the lot afterward